src/test_utils.jl

```diff
@@ -119,58 +136,75 @@ function StatsBase.fit(
 """
-function test_interface(template::Template; kwargs...)
+function test_interface(
+    template::Template; 
+    inputs=rand(5,5), 
+    outputs=_default_outputs(template),
+    distribution_inputs=[MvNormal(5, m) for m in 1:5],
+    kwargs...
+)
     @testset "Models API Interface Test: $(nameof(typeof(template)))" begin
-        test_interface(template, estimate_type(template), output_type(template); kwargs...)
+        predictions = test_common(template, inputs, outputs)
+        test_estimate_type(estimate_type(template), predictions, outputs)
+        test_output_type(output_type(template), predictions, outputs)
+        test_predict_input_type(predict_input_type(template), template, outputs, inputs, distribution_inputs)
     end
 end
 
-function test_interface(
-    template::Template, ::Type{PointEstimate}, ::Type{SingleOutput};
-    inputs=rand(5, 5), outputs=rand(1, 5),
-)
-    predictions = test_common(template, inputs, outputs)
-
-    @test predictions isa NamedDimsArray{(:variates, :observations), <:Real, 2}
-    @test size(predictions) == size(outputs)
-    @test size(predictions, 1) == 1
+function _default_outputs(template)
+    @assert(output_type(template) <: OutputTrait, "Invalid OutputTrait")
+    if output_type(template) == SingleOutput
+        return rand(1, 5)
+    else output_type(template) == MultiOutput
+        return rand(2, 5)
+    end
 end
 
-function test_interface(
-    template::Template, ::Type{PointEstimate}, ::Type{MultiOutput};
-    inputs=rand(5, 5), outputs=rand(2, 5),
-)
-    predictions = test_common(template, inputs, outputs)
+function test_estimate_type(::Type{PointEstimate}, predictions, outputs)
     @test predictions isa NamedDimsArray{(:variates, :observations), <:Real, 2}
     @test size(predictions) == size(outputs)
 end
 
-function test_interface(
-    template::Template, ::Type{DistributionEstimate}, ::Type{SingleOutput};
-    inputs=rand(5, 5), outputs=rand(1, 5),
-)
-    predictions = test_common(template, inputs, outputs)
-    @test predictions isa AbstractVector{<:ContinuousUnivariateDistribution}
+function test_estimate_type(::Type{DistributionEstimate}, predictions, outputs)
+    @test predictions isa AbstractVector{<:ContinuousDistribution}
     @test length(predictions) == size(outputs, 2)
-    @test all(length.(predictions) .== size(outputs, 1))
 end
 
-function test_interface(
-    template::Template, ::Type{DistributionEstimate}, ::Type{MultiOutput};
-    inputs=rand(5, 5), outputs=rand(3, 5)
-)
-    predictions = test_common(template, inputs, outputs)
-    @test predictions isa AbstractVector{<:ContinuousMultivariateDistribution}
-    @test length(predictions) == size(outputs, 2)
+function test_output_type(::Type{SingleOutput}, predictions, outputs)
     @test all(length.(predictions) .== size(outputs, 1))
+    @test all(length.(predictions) .== 1)
+end
+
+function test_output_type(::Type{MultiOutput}, predictions, outputs)
+    if  eltype(predictions) <: Distribution
```

Comment:
why use an `if eltype(x) <: T` check rather than dispatch? is it because we don't know what type of collection `predictions` is (e.g. it could be a Tuple or a AbstractVector or ...)?

Reply:
I hadn't thought of that but I think it might be hard to dispatch on.  The collection would be either a `NamedDimsArray{<:Real}` or an `AbstractVector{<:Distribution}`.  And the `NamedDimsArray` could be 2d.  So I think dispatching might be overly restrictive?

Reply:
yep, think the `eltype` check is a good call here